## Irvington Volunteer Project 

It's that time of year again! 

In your small town of Irvington, there is an annual festival that brings in visitors from all over the state! It's a huge deal and depends completely on the help of the local townspeople. 

The festival is 100% volunteer-run and operated, and it's your job to make sure all the volunteers have everything they need to get started. It is also your job to make sure volunteers are well organized, well prepared, and ready to rock when the day of the festival arrives! 

In the months before the festival, you collected a lot of information on each of the 1,000 volunteers. This information will help you better organize the volunteer effort. Follow the prompts below to complete your task of leading the volunteer effort!

***

#### For this task, use the Irvington Volunteers dataset. This dataset includes the following columns:

    * First = First name of volunteer
    * Last = Last name of volunteer
    * Gender = Gender of volunteer
    * Age = Age of volunteer         
    * Shift = Shift of volunteer worker; 1 = first shift; 2 = second shift; 3 = third shift; 4 = forth shift
    * Month = Month of year volunteer signed up for task
    * Volunteer Task = Assigned task for volunteer 
    * Task Level = Level of involvement for volunteer task; 1 = beginner, 2 = moderate, 3 = expert
    * Supervisor Number = Supervisor assigned to volunteer
    * Fees Owed = Outstanding volunteer fee's that are owed to community
    * Materials Needed = Does the volunteer still need task-specific materials; Y = yes; N = no
    * Volunteer Hours = Number of hours volunteer has spent preparing for task
    * Hours Pledged = Number of hours volunteer pledged to spend on task
    * Task Training Completed = Did the volunteer complete task-specific training; Y = yes; N = no
    * Recruited Volunteers = Number of other volunteers recruited by this volunteer

1. Import the "Irvington Volunteers" dataset. Check to make sure the dataset looks like what you're expecting

In [2]:
import pandas as pd
import numpy as np
df = pd.read_excel("Irvington_Volunteers.xlsx")

In [22]:
df.head()

First       Last Gender  Age  Shift  Month    Volunteer Task  Task Level  \
0  Jackie    Jackson      F   59      4     12       Fire Safety           2   
1    Mary  Patterson      F   31      4     10   First Aid Booth           1   
2   Tanya      Adams      F   48      1      3  Concession Stand           3   
3   Tanya  Henderson      F   19      2      2  Information Desk           3   
4  Walter   Franklin      M   25      1      2  Concession Stand           1   

   Supervisor Number  Fees Owed Materials Needed  Volunteer Hours  \
0                  1         18                Y               19   
1                  1          7                N               27   
2                  1         22                N               16   
3                  1         26                Y                9   
4                  1         25                Y               11   

   Hours Pledged Task Training Completed  Recruited Volunteers  
0             28                       Y                     3  
1             28                       N                     6  
2             28                       Y                     2  
3             28                       Y                     5  
4             28                       Y                     4

2. Confirm all your volunteers are accounted for and all of the information you need is within the dataset. What characteristics can you identify in the dataset?
***

* How many rows are in the dataset?
* How many columns are in the dataset?
* What types of data are within each column?

In [3]:
df.shape

(999, 15)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 999 entries, 0 to 998
Data columns (total 15 columns):
 #   Column                   Non-Null Count  Dtype 
---  ------                   --------------  ----- 
 0   First                    999 non-null    object
 1   Last                     999 non-null    object
 2   Gender                   999 non-null    object
 3   Age                      999 non-null    int64 
 4   Shift                    999 non-null    int64 
 5   Month                    999 non-null    int64 
 6   Volunteer Task           999 non-null    object
 7   Task Level               999 non-null    int64 
 8   Supervisor Number        999 non-null    int64 
 9   Fees Owed                999 non-null    int64 
 10  Materials Needed         999 non-null    object
 11  Volunteer Hours          999 non-null    int64 
 12  Hours Pledged            999 non-null    int64 
 13  Task Training Completed  999 non-null    object
 14  Recruited Volunteers     999 non-null    i

3. Before taking this list seriously, we need to check to make sure all the volunteers are eligible to work their specific task. The only eligibility requirement is age-based -- no volunteers can be under the age of 13 and no volunteers can be over the age of 75. Are there any volunteers that fall outside of this range?

In [5]:
df.max()

First                        india
Last                       Withers
Gender                           M
Age                             68
Shift                            4
Month                           12
Volunteer Task              Safety
Task Level                       3
Supervisor Number                5
Fees Owed                       30
Materials Needed                 Y
Volunteer Hours                 36
Hours Pledged                   28
Task Training Completed          Y
Recruited Volunteers             7
dtype: object

In [6]:
df.min()

First                             Aaron
Last                              Adams
Gender                                F
Age                                  16
Shift                                 1
Month                                 1
Volunteer Task             Booth Set-Up
Task Level                            1
Supervisor Number                     1
Fees Owed                             2
Materials Needed                      N
Volunteer Hours                       4
Hours Pledged                        28
Task Training Completed               N
Recruited Volunteers                  1
dtype: object

4. Did you miss anyone? Did you make sure to record ALL the information for ALL the volunteers? Quickly check if your missing any data in this dataset. 

In [7]:
df.isnull().sum()

First                      0
Last                       0
Gender                     0
Age                        0
Shift                      0
Month                      0
Volunteer Task             0
Task Level                 0
Supervisor Number          0
Fees Owed                  0
Materials Needed           0
Volunteer Hours            0
Hours Pledged              0
Task Training Completed    0
Recruited Volunteers       0
dtype: int64

5. A lot of the information we have organizes the volunteers into specific groups. It's important that none of these groups are mismatched (too many people in one group vs another). Let's check how many volunteers fall into each of the following groups. (Hint: you need to use the value counts function for this question). 
***

* Last (how many volunteers are from the same family?)
* Gender
* Shift
* Month
* Volunteer Task
* Task Level
* Supervisor Number
* Materials Needed
* Task Training Completed

In [8]:
df["Last"].value_counts()

Johnson      49
Turner       39
Carter       39
Michaels     38
Kennedy      38
Rose         38
Looner       38
Henderson    37
Martins      36
Davidson     35
Moore        34
Walters      34
Baker        34
Harrison     34
Kane         34
Withers      33
Patterson    33
Adams        33
Sapp         32
Vaughn       32
Rogers       31
Isaacson     31
Sears        30
Samuelson    29
Jackson      29
Gerardo      28
Franklin     28
Mulgrew      26
Ettienne     25
Monroe       22
Name: Last, dtype: int64

In [9]:
df["Gender"].value_counts()

M    510
F    489
Name: Gender, dtype: int64

In [10]:
df["Shift"].value_counts()

3    258
2    251
4    246
1    244
Name: Shift, dtype: int64

In [11]:
df["Month"].value_counts()

3     96
4     95
2     89
11    89
12    87
8     86
7     86
9     83
5     74
6     73
10    72
1     69
Name: Month, dtype: int64

In [12]:
df["Volunteer Task"].value_counts()

Concession Stand           142
Information Desk           131
First Aid Booth            130
Clean Up                   128
Safety                     124
Fire Safety                120
Booth Set-Up               114
Gardening and Lawn Care    110
Name: Volunteer Task, dtype: int64

In [13]:
df["Task Level"].value_counts()

3    352
2    342
1    305
Name: Task Level, dtype: int64

In [14]:
df["Supervisor Number"].value_counts()

3    212
2    207
5    202
4    195
1    183
Name: Supervisor Number, dtype: int64

In [15]:
df["Materials Needed"].value_counts()

Y    637
N    362
Name: Materials Needed, dtype: int64

In [16]:
df["Task Training Completed"].value_counts()

Y    586
N    413
Name: Task Training Completed, dtype: int64

6. Let's look at these same groups, but from another angle. It's important that the community members are mixing and mingling with members of all ages. What is the average age of volunteers within all the groups listed below? (Hint: you need to use the groupby function for this question). Once you determine the average age of volunteers in the groups below, determine the average number of volunteer hours for all the groups below. 
***

* Gender
* Shift
* Month
* Volunteer Task
* Task Level
* Supervisor Number
* Materials Needed
* Task Training Completed

In [17]:
df.groupby(df['Age']).mean()

Shift     Month  Task Level  Supervisor Number  Fees Owed  \
Age                                                                 
16   2.333333  6.047619    2.190476           3.047619  17.380952   
17   2.550000  6.800000    2.200000           3.200000  16.850000   
18   2.000000  6.066667    2.333333           3.200000  16.800000   
19   2.214286  5.571429    2.357143           2.714286  17.357143   
20   2.285714  6.952381    2.047619           3.142857  19.714286   
21   2.578947  7.157895    2.105263           2.263158  17.947368   
22   2.500000  6.812500    2.125000           3.000000  14.375000   
23   2.631579  7.526316    1.789474           2.947368  19.526316   
24   3.000000  6.684211    1.947368           2.736842  16.684211   
25   2.600000  6.600000    2.066667           2.666667  13.133333   
26   2.600000  6.850000    2.400000           3.050000  15.700000   
27   2.166667  6.611111    2.277778           2.944444  16.166667   
28   2.142857  7.714286    2.142857           3.000000  19.142857   
29   2.680000  6.680000    1.880000           3.200000  16.480000   
30   2.375000  5.125000    2.125000           3.062500  16.375000   
31   2.428571  6.428571    1.809524           3.333333  18.619048   
32   2.133333  5.200000    2.133333           2.333333  13.333333   
33   2.272727  5.818182    2.409091           2.909091  15.727273   
34   3.142857  6.142857    2.428571           3.142857  18.000000   
35   2.476190  7.666667    2.000000           2.523810  18.666667   
36   2.238095  7.285714    1.952381           2.809524  13.952381   
37   2.739130  5.956522    1.956522           3.434783  13.304348   
38   2.360000  5.200000    1.960000           3.240000  13.080000   
39   2.590909  7.045455    1.772727           3.227273  16.227273   
40   2.666667  5.722222    2.000000           2.777778  11.666667   
41   3.083333  6.916667    2.083333           2.500000  18.583333   
42   2.857143  3.500000    1.642857           3.285714  13.928571   
43   2.750000  5.250000    2.300000           2.650000  13.350000   
44   2.526316  6.526316    1.736842           2.368421  17.578947   
45   2.066667  7.333333    2.200000           3.133333  18.333333   
46   2.541667  6.125000    2.041667           3.541667  14.458333   
47   2.400000  5.800000    2.400000           2.500000  15.700000   
48   2.368421  6.263158    2.000000           3.263158  16.421053   
49   2.350000  6.900000    2.000000           3.300000  16.650000   
50   2.633333  5.633333    1.933333           3.433333  14.700000   
51   2.315789  6.894737    1.736842           2.473684  13.473684   
52   3.200000  6.533333    2.066667           2.666667  16.533333   
53   2.800000  8.000000    1.950000           3.500000  18.250000   
54   2.958333  7.458333    1.750000           3.541667  19.166667   
55   2.133333  7.400000    1.666667           2.333333  16.933333   
56   2.266667  7.000000    2.266667           3.666667  18.200000   
57   2.454545  6.681818    1.818182           3.136364  13.863636   
58   2.560000  7.160000    2.280000           2.600000  16.440000   
59   2.214286  6.928571    2.071429           3.357143  14.857143   
60   2.600000  7.333333    1.733333           3.200000  11.333333   
61   2.434783  6.565217    2.347826           2.739130  18.565217   
62   2.450000  6.600000    1.850000           3.000000  20.000000   
63   2.592593  5.333333    2.148148           3.518519  15.222222   
64   2.266667  7.133333    1.933333           3.400000  15.600000   
65   2.842105  7.052632    1.842105           3.000000  19.052632   
66   2.272727  6.636364    2.272727           2.272727  18.454545   
67   2.807692  6.653846    2.192308           2.923077  15.692308   
68   2.166667  5.916667    2.333333           3.541667  17.291667   

     Volunteer Hours  Hours Pledged  Recruited Volunteers  
Age                                                        
16         18.238095           28.0              4.380952  
17         19.200000           2

In [20]:
df.groupby(df['Volunteer Hours']).mean()

Age     Shift      Month  Task Level  \
Volunteer Hours                                               
4                35.000000  3.000000  11.000000    2.000000   
5                43.727273  2.454545   6.818182    1.727273   
6                36.933333  2.533333   4.800000    2.066667   
7                40.347826  2.869565   6.652174    2.043478   
8                42.322581  2.548387   7.548387    2.129032   
9                43.833333  2.666667   6.547619    2.095238   
10               43.852941  2.411765   7.058824    2.176471   
11               39.588235  2.000000   6.735294    2.029412   
12               45.837838  2.513514   7.054054    2.135135   
13               43.829268  2.390244   6.878049    2.073171   
14               41.756757  2.459459   5.783784    1.756757   
15               41.562500  2.500000   6.531250    1.906250   
16               42.770833  2.375000   6.520833    2.208333   
17               43.812500  2.656250   6.562500    2.156250   
18               41.975610  2.512195   5.951220    2.048780   
19               42.194444  2.694444   7.083333    2.305556   
20               37.129032  2.645161   6.548387    1.967742   
21               41.896552  2.241379   7.344828    1.827586   
22               42.717949  2.435897   5.974359    2.230769   
23               43.777778  2.777778   6.111111    1.861111   
24               44.342857  2.600000   7.000000    2.057143   
25               43.540000  2.560000   5.820000    1.860000   
26               44.028571  2.514286   6.571429    2.000000   
27               40.225806  2.451613   5.967742    2.225806   
28               43.411765  2.411765   6.294118    1.970588   
29               43.893617  2.255319   6.638298    2.234043   
30               35.718750  2.781250   6.437500    2.062500   
31               43.135135  2.540541   6.486486    1.837838   
32               42.840000  2.360000   6.160000    2.280000   
33               43.950000  2.700000   7.000000    1.850000   
34               50.400000  2.333333   5.666667    1.733333   
35               42.000000  3.166667   7.000000    2.500000   
36               31.500000  2.500000   6.500000    2.000000   

                 Supervisor Number  Fees Owed  Hours Pledged  \
Volunteer Hours                                                
4                         5.000000  30.000000           28.0   
5                         2.909091  29.818182           28.0   
6                         3.066667  29.133333           28.0   
7                         2.913043  28.521739           28.0   
8                         3.483871  28.483871           28.0   
9                         2.476190  27.785714           28.0   
10                        3.205882  26.794118           28.0   
11                        3.029412  25.323529           28.0   
12                        3.108108  24.729730           28.0   
13                        3.170732  23.097561           28.0   
14                        2.918919  21.918919           28.0   
15                        2.968750  21.406250           28.0   
16                        2.833333  20.125000           28.0   
17                        2.937500  19.187500           28.0   
18                        3.195122  18.268293           28.0   
19                        3.138889  17.138889           28.0   
20                        2.806452  15.709677           28.0   
21                        2.689655  15.413793           28.0   
22                        3.205128  13.871795           28.0   
23                        3.194444  13.138889           28.0   
24                        3.057143  12.057143           28.0   
25                        2.820000  10.680000           28.0   
26                        3.085714   9.914286           28.0   
27                        2.806452   8.258065           28.0   
28                        3.176471   7.941176           28.0   
29                        3.191489   6.787234           28.0   
30                       

7. There are specific groups of volunteers that need to be followed-up with immediately! These are the volunteers that have some outstanding business to handle before the big festival. Locate the group of volunteers that fall within each of these conditions.
***
* Locate the volunteers who still need materials for their volunteer task
* Locate the volunteers who have not yet completed their volunteer training
* Locate the volunteers who are experts are their task (aka Task Level = 3)
* Locate the volunteers who are working fire safety and haven't yet completed their training - these folks need to get trained ASAP!

In [23]:
df.loc[df["Materials Needed"] == "Y"]

First       Last Gender  Age  Shift  Month           Volunteer Task  \
0      Jackie    Jackson      F   59      4     12              Fire Safety   
3       Tanya  Henderson      F   19      2      2         Information Desk   
4      Walter   Franklin      M   25      1      2         Concession Stand   
6      Martin   Ettienne      M   36      2      8  Gardening and Lawn Care   
8      Denise  Henderson      F   38      1      6          First Aid Booth   
..        ...        ...    ...  ...    ...    ...                      ...   
993    Walter    Martins      M   67      3     12              Fire Safety   
994    Walter    Mulgrew      M   63      3      4         Information Desk   
996     Larry  Patterson      M   49      1      4                 Clean Up   
997    Victor       Kane      M   39      3      7         Concession Stand   
998  Victoria     Rogers      F   29      4      4  Gardening and Lawn Care   

     Task Level  Supervisor Number  Fees Owed Materials Needed  \
0             2                  1         18                Y   
3             3                  1         26                Y   
4             1                  1         25                Y   
6             2                  1          8                Y   
8             2                  1         12                Y   
..          ...                ...        ...              ...   
993           1                  5          9                Y   
994           2                  5         10                Y   
996           2                  5         21                Y   
997           2                  5         28                Y   
998           2                  5         21                Y   

     Volunteer Hours  Hours Pledged Task Training Completed  \
0                 19             28                       Y   
3                  9             28                       Y   
4                 11             28                       Y   
6                 26             28                       N   
8                 23             28                       N   
..               ...            ...                     ...   
993               28             28                       N   
994               25             28                       Y   
996               16             28                       N   
997                8             28                       Y   
998               18             28                       N   

     Recruited Volunteers  
0                       3  
3                       5  
4                       4  
6                       6  
8                       5  
..                    ...  
993                     3  
994                     5  
996                     3  
997                     4  
998                     1  

[637 rows x 15 columns]

In [24]:
df.loc[df["Task Training Completed"] == "N"]

First       Last Gender  Age  Shift  Month           Volunteer Task  \
1        Mary  Patterson      F   31      4     10          First Aid Booth   
6      Martin   Ettienne      M   36      2      8  Gardening and Lawn Care   
7       Peter   Michaels      M   36      1      6         Concession Stand   
8      Denise  Henderson      F   38      1      6          First Aid Booth   
24      Roger   Walters       M   61      2      7                 Clean Up   
..        ...        ...    ...  ...    ...    ...                      ...   
986       Sam    Withers      M   37      4     11  Gardening and Lawn Care   
989     Onika      Sears      F   38      1      1         Concession Stand   
993    Walter    Martins      M   67      3     12              Fire Safety   
996     Larry  Patterson      M   49      1      4                 Clean Up   
998  Victoria     Rogers      F   29      4      4  Gardening and Lawn Care   

     Task Level  Supervisor Number  Fees Owed Materials Needed  \
1             1                  1          7                N   
6             2                  1          8                Y   
7             3                  1          7                N   
8             2                  1         12                Y   
24            3                  1         25                Y   
..          ...                ...        ...              ...   
986           2                  5         22                N   
989           1                  5          3                N   
993           1                  5          9                Y   
996           2                  5         21                Y   
998           2                  5         21                Y   

     Volunteer Hours  Hours Pledged Task Training Completed  \
1                 27             28                       N   
6                 26             28                       N   
7                 27             28                       N   
8                 23             28                       N   
24                12             28                       N   
..               ...            ...                     ...   
986               16             28                       N   
989               34             28                       N   
993               28             28                       N   
996               16             28                       N   
998               18             28                       N   

     Recruited Volunteers  
1                       6  
6                       6  
7                       6  
8                       5  
24                      3  
..                    ...  
986                     2  
989                     3  
993                     3  
996                     3  
998                     1  

[413 rows x 15 columns]

In [25]:
df.loc[df["Task Level"] == 3]

First       Last Gender  Age  Shift  Month    Volunteer Task  \
2     Tanya      Adams      F   48      1      3  Concession Stand   
3     Tanya  Henderson      F   19      2      2  Information Desk   
7     Peter   Michaels      M   36      1      6  Concession Stand   
10      Sam   Michaels      M   57      4      3   First Aid Booth   
11     Jada   Walters       F   20      1      5  Concession Stand   
..      ...        ...    ...  ...    ...    ...               ...   
980  August    Jackson      F   30      1     11          Clean Up   
981   Betty     Vaughn      F   65      2      3            Safety   
982    Mary    Kennedy      F   36      3     11  Information Desk   
985  Martin    Martins      M   34      2     12  Information Desk   
988    Mary     Carter      F   60      3      1  Information Desk   

     Task Level  Supervisor Number  Fees Owed Materials Needed  \
2             3                  1         22                N   
3             3                  1         26                Y   
7             3                  1          7                N   
10            3                  1          5                Y   
11            3                  1         26                N   
..          ...                ...        ...              ...   
980           3                  5         26                N   
981           3                  5         28                N   
982           3                  5         15                Y   
985           3                  5         21                N   
988           3                  5         25                Y   

     Volunteer Hours  Hours Pledged Task Training Completed  \
2                 16             28                       Y   
3                  9             28                       Y   
7                 27             28                       N   
10                32             28                       Y   
11                10             28                       Y   
..               ...            ...                     ...   
980               11             28                       Y   
981               10             28                       N   
982               20             28                       N   
985               13             28                       Y   
988               10             28                       Y   

     Recruited Volunteers  
2                       2  
3                       5  
7                       6  
10                      3  
11                      4  
..                    ...  
980                     3  
981                     2  
982                     5  
985                     6  
988                     5  

[352 rows x 15 columns]

In [28]:
df.loc[(df["Volunteer Task"] == "Fire Safety") & (["Task Training Completed"] == "N")]

Empty DataFrame
Columns: [First, Last, Gender, Age, Shift, Month, Volunteer Task, Task Level, Supervisor Number, Fees Owed, Materials Needed, Volunteer Hours, Hours Pledged, Task Training Completed, Recruited Volunteers]
Index: []

8. To keep track of volunteers who meet specific conditions, make some additional columns to hold the new information. 
***

* Inexperienced volunteers may get confused or lost. Create a new column called "needsMentor". This column should be assigned a value of "1" if a volunteer has not completed training and is at skill level 1. 
* How many volunteers haven't completed the volunteer hours they promised they would? Create a new column that subtracts the number of hours pledged ("Hours Pledged") from the number of volunteer hours already worked ("Volunteer Hours). Call this column "hoursNeeded". 
* There are some overly committed volunteers - they have already worked all their pledged volunteer hours and are still going! These folks need to get a specific bonus when the festival is over - we need to track them somehow. Create a new column called "overtimeBonus" - if "hoursNeeded" is less than 0, this column should be "1". 

In [30]:
df["needsMentor"] = np.where(((df["Task Training Completed"] == "N") & (df["Task Level"] == 1)), 1, 0)


In [31]:
df["hoursNeeded"] = df["Volunteer Hours"] - df["Hours Pledged"]

In [32]:
df["overtimeBonus"] = np.where(df["hoursNeeded"] < 0, 1, 0)

9. The Mayor wants to make a speech about the different types of community members that are volunteering for the festival. Specifically, the Mayor would like to present some information on the different age groups. Create a new column called "Age Group" and bin the "Age" column to create bins based on the age of the volunteer. Follow the guidance below for the variations in age. (Hint: you can use any code you like to complete this task - np.select is also an option!)
***

* Teen (0 - 17)
* Young Adult (17.1 - 35)
* Adult (35.1 - 65)
* Senior (65+)

In [33]:
bins = [0, 17, 35, 65, 150]
bin_labels = ["Teen", "Young Adult", "Adult", "Senior"]
df["Ages"] = pd.cut(df["Age"], bins, labels = bin_labels)

10. The Supervisor Number column is all messed up. Someone didn't record is correctly, and now it is meaningless! Drop it from this dataset - it's not providing any meaningful information. 

In [34]:
df.drop(columns = "Supervisor Number", inplace = True)

11. There is a new policy about volunteer fees. It seems that some of the fees owed were actually for materials that the volunteer paid for out of pocket. Wipe out the debt for those volunteers who owe just a few dollars. If the volunteer owes less than 5 dollars, replace that value with 0!

In [37]:
df["Fees Owed"].replace(1, 0, inplace = True)

In [38]:
df["Fees Owed"].replace(2, 0, inplace = True)

In [39]:
df["Fees Owed"].replace(3, 0, inplace = True)

In [40]:
df["Fees Owed"].replace(4, 0, inplace = True)

In [41]:
df["Fees Owed"].replace(5, 0, inplace = True)

12. We need some code that we can keep around for next year -- when we have the same task but new data! Define a few functions that can complete tasks that we will need to repeat down the road. 

***

* Define a function to recode the column "Gender". Instead of "M" and "F" -- have the words spelled out "Male" and "Female". Apply this function to the "Gender" column. 
* Define a function to recode the column "Month". Convert the numeric "Month" to the name of the Month. Apply this function to the column "Month" and create a new column called "Name of Month". 

In [35]:
def gender(gen):
    if gen == "M":
        return "Male"
    else:
        return "Female" 

df["Gender"] = df["Gender"].apply(gender)

df

First       Last  Gender  Age  Shift  Month           Volunteer Task  \
0      Jackie    Jackson  Female   59      4     12              Fire Safety   
1        Mary  Patterson  Female   31      4     10          First Aid Booth   
2       Tanya      Adams  Female   48      1      3         Concession Stand   
3       Tanya  Henderson  Female   19      2      2         Information Desk   
4      Walter   Franklin    Male   25      1      2         Concession Stand   
..        ...        ...     ...  ...    ...    ...                      ...   
994    Walter    Mulgrew    Male   63      3      4         Information Desk   
995  Veronica     Looner  Female   63      3      5                 Clean Up   
996     Larry  Patterson    Male   49      1      4                 Clean Up   
997    Victor       Kane    Male   39      3      7         Concession Stand   
998  Victoria     Rogers  Female   29      4      4  Gardening and Lawn Care   

     Task Level  Fees Owed Materials Needed  Volunteer Hours  Hours Pledged  \
0             2         18                Y               19             28   
1             1          7                N               27             28   
2             3         22                N               16             28   
3             3         26                Y                9             28   
4             1         25                Y               11             28   
..          ...        ...              ...              ...            ...   
994           2         10                Y               25             28   
995           2         28                N                8             28   
996           2         21                Y               16             28   
997           2         28                Y                8             28   
998           2         21                Y               18             28   

    Task Training Completed  Recruited Volunteers  needsMentor  hoursNeeded  \
0                         Y                     3            0           -9   
1                         N                     6            1           -1   
2                         Y                     2            0          -12   
3                         Y                     5            0          -19   
4                         Y                     4            0          -17   
..                      ...                   ...          ...          ...   
994                       Y                     5            0           -3   
995                       Y                     4            0          -20   
996                       N                     3            0          -12   
997                       Y                     4            0          -20   
998                       N                     1            0          -10   

     overtimeBonus         Ages  
0                1        Adult  
1                1  Young Adult  
2                1        Adult  
3                1  Young Adult  
4                1  Young Adult  
..             ...          ...  
994              1        Adult  
995              1        Adult  
996              1        Adult  
997              1        Adult  
998              1  Young Adult  

[999 rows x 18 columns]

In [36]:
def months(month):
    if month == 1:
        return "january"
    elif month == 2:
        return "february"
    elif month == 3:
        return "march"
    elif month == 4:
        return "april"
    elif month == 5:
        return "may"
    elif month == 6:
        return "june"
    elif month == 7:
        return "july"
    elif month == 8:
        return "august"
    elif month == 9:
        return "september"
    elif month == 10:
        return "october"
    elif month == 11:
        return "november"
    else:
        return "december" 

df["Month"] = df["Month"].apply(months)

df

First       Last  Gender  Age  Shift     Month  \
0      Jackie    Jackson  Female   59      4  december   
1        Mary  Patterson  Female   31      4   october   
2       Tanya      Adams  Female   48      1     march   
3       Tanya  Henderson  Female   19      2  february   
4      Walter   Franklin    Male   25      1  february   
..        ...        ...     ...  ...    ...       ...   
994    Walter    Mulgrew    Male   63      3     april   
995  Veronica     Looner  Female   63      3       may   
996     Larry  Patterson    Male   49      1     april   
997    Victor       Kane    Male   39      3      july   
998  Victoria     Rogers  Female   29      4     april   

              Volunteer Task  Task Level  Fees Owed Materials Needed  \
0                Fire Safety           2         18                Y   
1            First Aid Booth           1          7                N   
2           Concession Stand           3         22                N   
3           Information Desk           3         26                Y   
4           Concession Stand           1         25                Y   
..                       ...         ...        ...              ...   
994         Information Desk           2         10                Y   
995                 Clean Up           2         28                N   
996                 Clean Up           2         21                Y   
997         Concession Stand           2         28                Y   
998  Gardening and Lawn Care           2         21                Y   

     Volunteer Hours  Hours Pledged Task Training Completed  \
0                 19             28                       Y   
1                 27             28                       N   
2                 16             28                       Y   
3                  9             28                       Y   
4                 11             28                       Y   
..               ...            ...                     ...   
994               25             28                       Y   
995                8             28                       Y   
996               16             28                       N   
997                8             28                       Y   
998               18             28                       N   

     Recruited Volunteers  needsMentor  hoursNeeded  overtimeBonus  \
0                       3            0           -9              1   
1                       6            1           -1              1   
2                       2            0          -12              1   
3                       5            0          -19              1   
4                       4            0          -17              1   
..                    ...          ...          ...            ...   
994                     5            0           -3              1   
995                     4            0          -20              1   
996                     3            0          -12              1   
997                     4            0          -20              1   
998                     1            0          -10              1   

            Ages  
0          Adult  
1    Young Adult  
2          Adult  
3    Young Adult  
4    Young Adult  
..           ...  
994        Adult  
995        Adult  
996        Adult  
997        Adult  
998  Young Adult  

[999 rows x 18 columns]